src/services/posthog/telemetry/TelemetryService.ts

```diff
@@ -129,40 +116,72 @@ class PostHogClient {
 		this.client = posthogClientProvider.getClient()
 	}
 
+	private setDistinctId(installId: string) {
+		if (this.distinctId === "someValue.machineId") {
```

Comment:
Based on my testing in this branch, and some looking at the code here.  This looks like it will never be a true condition.  Can you explain how this works and where we would ever match true for distinctId to be `someValue.machineId`

Reply:
OK.  We should try to find out what actually causes this and make a comment about that.  I'll approve for now, but please follow up with a comment explaining why this is needed, so it's not ripped out in the future.